hazelcast/src/main/java/com/hazelcast/internal/partition/impl/PartitionReplicaManager.java

```diff
@@ -329,7 +348,7 @@ public void clearReplicaSyncRequest(int partitionId, ServiceNamespace namespace,
             logger.finest("Clearing sync replica request for partitionId=" + partitionId + ", replicaIndex="
                     + replicaIndex + ", namespace=" + namespace);
         }
-        releaseReplicaSyncPermit();
+        releaseReplicaSyncPermits(1);
         replicaSyncTimeoutScheduler.cancelIfExists(syncInfo, null);
```

Comment:
I guess there might be some unfortunate timing that may cause a permit to be released but the task to be executed as well. But it's a minor issue.

Reply:
We release a permit only when `replicaSyncRequests.remove(syncInfo)` succeeds, so there won't be multiple permit releases.